src/main/java/dev/jbang/source/SourceSet.java

```diff
@@ -12,94 +12,56 @@
 import dev.jbang.Settings;
 import dev.jbang.dependencies.DependencyResolver;
 import dev.jbang.dependencies.MavenRepo;
-import dev.jbang.dependencies.ModularClassPath;
 import dev.jbang.source.generators.JarCmdGenerator;
 import dev.jbang.source.generators.JshCmdGenerator;
-import dev.jbang.source.resolvers.SiblingResourceResolver;
-import dev.jbang.util.JavaUtil;
 import dev.jbang.util.Util;
 
 public class SourceSet implements Code {
-	private final ResourceResolver resolver;
-	private final List<Source> sources = new ArrayList<>();
+	private final Source mainSource;
+
+	private final List<ResourceRef> sources = new ArrayList<>();
 	private final List<RefTarget> resources = new ArrayList<>();
 	private final List<String> dependencies = new ArrayList<>();
 	private final List<MavenRepo> repositories = new ArrayList<>();
 	private final List<String> classPaths = new ArrayList<>();
 	private final List<String> compileOptions = new ArrayList<>();
 	private final List<String> runtimeOptions = new ArrayList<>();
-	private final List<KeyValue> agentOptions = new ArrayList<>();
+	private final Map<String, String> manifestAttributes = new LinkedHashMap<>();
 	private String javaVersion;
 	private String description;
 	private String gav;
 	private String mainClass;
 
-	private ModularClassPath mcp;
-	private File jarFile;
+	private Path jarFile;
 	private Jar jar;
 
-	public static SourceSet forSource(Source mainSource) {
-		return new SourceSet(mainSource, ResourceResolver.forResources());
-	}
-
-	public static SourceSet forSource(Source mainSource, ResourceResolver resolver) {
-		return new SourceSet(mainSource, resolver);
-	}
+	public static final String ATTR_PREMAIN_CLASS = "Premain-Class";
+	public static final String ATTR_AGENT_CLASS = "Agent-Class";
+	public static final String ATTR_STAGE_PREBUILD_CLASS = "JBang-Stage-Prebuild-Class";
```

Comment:
this does not look like a refactoring ?

Reply:
Oops, I hadn't noticed that one. A line that shouldn't be there.